app/components/trainee_status_card/view.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module TraineeStatusCard
+  class View < GovukComponent::Base
+    attr_reader :state, :user, :target
+
+    STATUS_COLOURS = {
+      draft: "grey",
+      submitted_for_trn: "turquoise",
+      trn_received: "blue",
+      recommended_for_qts: "purple",
+      qts_awarded: "",
+      deferred: "yellow",
+      withdrawn: "red",
+    }.freeze
+
+    def initialize(state:, user:, target:)
+      @state = state
+      @user = user
+      @target = target
+    end
+
+    def count
+      if user.system_admin?
```

Comment:
It feels like this sort of code (fetching the correct data to display) should be in the controller and not in the view code itself.

If you fetch the trainees in the controller, you can then let the policy (via pundit) decide which trainees the user should see i.e.

`@trainees = policy_scope(Trainee.all)`

Then this 'smart' component can be passed in a collection of `@trainees` (and not the user):

```
<%= render TraineeStatusCard::View.new(state: :draft, target: target_path, trainees: @trainees) %>
```

And apply the filtering, not caring about the user policy:

```
def count
  trainees.where(state: state).count
end
```

Reply:
This is really interesting, so it's basically the resolve method in the scope class within trainee policy which does the work of sorting which trainees are shown (either all of them if sys admin or just the ones associated with the user's provider if not sys admin) ?

Reply:
you got it!